.github/workflows/build_and_publish_call.yaml

```diff
@@ -0,0 +1,99 @@
+name: Build and Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      ubuntu-version:
+        description: "Ubuntu version for the base image"
+        default: "22.04"
+        type: string
+        required: false
+      mlflow-version:
+        description: "MLflow pip library version to use during build of the image"
+        default: "2.1.1"
+        type: string
+        required: false
+      version:
+        description: "Current image version for given MLflow and Ubuntu versions"
+        default: "1.0"
+        type: string
+        required: false
+    secrets:
+      DOCKER_HUB_CHARMED_USERNAME:
+        required: true
+      DOCKER_HUB_CHARMED_ACCESS_TOKEN:
+        required: true
+  workflow_call:
+    inputs:
+      ubuntu-version:
+        description: "Ubuntu version for the base image"
+        default: "22.04"
+        type: string
+        required: false
+      mlflow-version:
+        description: "MLflow pip library version to use during build of the image"
+        default: "2.1.1"
+        type: string
+        required: false
+      version:
+        description: "Current image version for given MLflow and Ubuntu versions"
+        default: "1.0"
+        type: string
+        required: false
+    secrets:
+      DOCKER_HUB_CHARMED_USERNAME:
+        required: true
+      DOCKER_HUB_CHARMED_ACCESS_TOKEN:
+        required: true
+
+jobs:
+  build-and-push-docker-image:
+    name: Build Docker image and push to repositories
+    permissions:
+      id-token: write
+      contents: write
+      security-events: write
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      
+      - name: Build Docker image
+        run: |
+          docker build -t ${{ vars.DOCKER_HUB_CHARMED_IMAGE }}:${{ github.sha }} --build-arg VERSION=${{ inputs.mlflow-version }} --build-arg UBUNTU_VERSION=${{ inputs.ubuntu-version }} .
```

Comment:
should not we build multi arch ?

Reply:
Dont understand ... like 22.04 and 20.04 ?